### DISCUSSION TITLE
Reading .msh file issue

### DISCUSSION THREAD

**Discussion by hamilton-eng** [ansys/pyprimemesh#720] [Q&A]
Having issues reading msh file, not sure if this is a compatibility issue? what sort of .msh files can be used by the api?

using a test .msh file from url below

[https://github.com/amandaghassaei/msh-parser/blob/main/test/msh/stanford_bunny.msh](url)

```
from ansys.meshing import prime
from ansys.meshing.prime import lucid

prime_client= prime.launch_prime()
model= prime_client.model


    

mesher= lucid.Mesh(model)
mesh_file_name= r"tests/stanford_bunny.msh"
mesher.read(mesh_file_name, append=False)
cdb_file_name= r"data/output/cdb_file.cdb"
mesher.write(cdb_file_name)
```

`Traceback (most recent call last):
  File "c:\Users\User\tests\Ansys_geom.py", line 12, in <module>
    mesher.read(mesh_file_name, append=False)
  File "C:\Users\User\miniconda3\Lib\site-packages\ansys\meshing\prime\lucid\mesh_util.py", line 76, in read
    prime.FileIO(self._model).import_fluent_meshing_meshes(
  File "C:\Users\User\miniconda3\Lib\site-packages\ansys\meshing\prime\core\fileio.py", line 304, in import_fluent_meshing_meshes   
    result = super().import_fluent_meshing_meshes(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\miniconda3\Lib\site-packages\ansys\meshing\prime\autogen\fileio.py", line 352, in import_fluent_meshing_meshes
    result = self._comm.serve(self._model, command_name, self._object_id, args=args)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\miniconda3\Lib\site-packages\ansys\meshing\prime\internals\error_handling.py", line 515, in wrapper_error_code
    raise PrimeRuntimeError(
ansys.meshing.prime.internals.error_handling.PrimeRuntimeError: Reading mesh failed.`

**Comment by hamilton-eng** [comment]
I'm also trying to import .STL

```

import ansys.meshing.prime as prime
from ansys.meshing.prime.graphics import Graphics

prime_client = prime.launch_prime()
model = prime_client.model
mesh_util = prime.lucid.Mesh(model=model)



input_file = r'tests/plane.stl'

file_io=prime.FileIO(model)
file_io.import_cad(
    input_file,
    params=prime.ImportCadParams(model=model, length_unit=prime.LengthUnit.MM),
)

# display the rear wing geometry without the enclosure
display = Graphics(model)
display(update=True)
part= model.get_part_by_name('plane')
part_summmary_res = part.get_summary(prime.PartSummaryParams(model=model))
```
Error


C:\Users\User\miniconda3\Lib\site-packages\pyvista\plotting\plotting\__init__.py:23: PyVistaDeprecationWarning: The `pyvista.plotting.plotting` module has been deprecated. `Plotter` is now imported as: `from pyvista.plotting import Plotter`.
  warnings.warn(
Traceback (most recent call last):
  File "c:\Users\User\tests\Ansys_geom.py", line 25, in <module>
    file_io.import_cad(
  File "C:\Users\User\miniconda3\Lib\site-packages\ansys\meshing\prime\core\fileio.py", line 472, in import_cad
    import_result = super().import_cad(temp_file_name, params)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\miniconda3\Lib\site-packages\ansys\meshing\prime\autogen\fileio.py", line 535, in import_cad
    result = self._comm.serve(self._model, command_name, self._object_id, args=args)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\miniconda3\Lib\site-packages\ansys\meshing\prime\internals\error_handling.py", line 515, in wrapper_error_code
    raise PrimeRuntimeError(
ansys.meshing.prime.internals.error_handling.PrimeRuntimeError: CAD import failed.


I'm using 0.4 documentation and have 0.4 pyprimemesh with 2023 R2 Academic

**Comment by waltersma** [comment]
Hi @hamilton-eng,

The stanford_bunny.msh is not a Fluent .msh format.  Currently we do not support non Fluent .msh formats. 

Regarding the stl import can you verify if stl import is working for other files or if this is file specific? 

You could try the following for example:

`input_file = prime.examples.download_f1_rw_drs_stl()`

I hope this helps!

**Reply by hamilton-eng** [reply]
```import ansys.meshing.prime as prime
from ansys.meshing.prime.graphics import Graphics

prime_client = prime.launch_prime()
model = prime_client.model




#input_file = r'tests/plane.stl'




input_file = prime.examples.download_f1_rw_drs_stl()
file_io=prime.FileIO(model)
file_io.import_cad(
    input_file,
    params=prime.ImportCadParams(model=model, length_unit=prime.LengthUnit.MM),
)
display = Graphics(model)
display(update=True)```


C:\Users\User\miniconda3\Lib\site-packages\pyvista\plotting\plotting\__init__.py:23: PyVistaDeprecationWarning: The `pyvista.plotting.plotting` module has been deprecated. `Plotter` is now imported as: `from pyvista.plotting import Plotter`.
  warnings.warn(
Traceback (most recent call last):
  File "c:\Users\User\Documents\Zara\Regolith_Twinning\tests\Ansys_geom.py", line 26, in <module>
    file_io.import_cad(
  File "C:\Users\User\miniconda3\Lib\site-packages\ansys\meshing\prime\core\fileio.py", line 472, in import_cad
    import_result = super().import_cad(temp_file_name, params)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\miniconda3\Lib\site-packages\ansys\meshing\prime\autogen\fileio.py", line 535, in import_cad
    result = self._comm.serve(self._model, command_name, self._object_id, args=args)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\miniconda3\Lib\site-packages\ansys\meshing\prime\internals\error_handling.py", line 515, in wrapper_error_code
    raise PrimeRuntimeError(
ansys.meshing.prime.internals.error_handling.PrimeRuntimeError: CAD import failed.

**Comment by waltersma** [comment]
Hi @hamilton-eng ,

Please can you check that the environment variable AWP_ROOT232 is set correctly to your Ansys installation directory?  The path should end like "\ANSYS Inc\v232" for example.

This should be set during installation but please confirm this is setup.

Thanks

**Reply by hamilton-eng** [reply]
Thank you, yes it appears to be in the directory

C:\Users\User>echo %AWP_ROOT232%
C:\Program Files\ANSYS Inc\ANSYS Student\v232

**Comment by waltersma** [comment]
Hi @hamilton-eng 

Please share the resulting log file from the temp folder (\AppData\Local\Temp) so that we can try and see what the issue is?

Thanks!

**Reply by hamilton-eng** [reply]
[ansyscl.DESKTOP-VMGC4IP.17148.53298.log](https://github.com/ansys/pyprimemesh/files/14450237/ansyscl.DESKTOP-VMGC4IP.17148.53298.log)

**Reply by waltersma** [reply]
Hi @hamilton-eng,

Thank you for your help on this issue.  
If it is possible please could I ask you to send some additional files for debugging?  There should be the following two additional files:  

1. ansyscl.DESKTOP-VMGC4IP.log
2. licdebug*ANSYS PRIME*

Thanks!
